packages/instant/src/components/animations/slide_up_and_down_animation.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+
+import { keyframes, styled } from '../../style/theme';
+
+const slideKeyframeGenerator = (fromY: string, toY: string) => keyframes`
+    from {
+        position: relative;
+        top: ${fromY}
+    }
+
+    to {
+        position: relative;
+        top: ${toY}
+    }
+`;
+
+export interface SlideAnimationProps {
+    keyframes: string;
+    animationType: string;
+    animationDirection?: string;
+}
+export const SlideAnimation =
+    styled.div <
+    SlideAnimationProps >
+    `
```

Comment:
What is this syntax?? 🆒

Reply:
Also why not just: `export const SlideAnimation = styled.div<SlideAnimationProps>`? Are you using `prettier`?

Reply:
VSCode prettier integration seems to be autoformatting it like this:

![prettier](https://user-images.githubusercontent.com/210914/46967474-78b1e980-d065-11e8-9133-29557a39e885.gif)

Reply:
hm, strange 🤷‍♂️

Reply:
OH I didn't know you could do this. Cool.